review/src/PTUI.java

```diff
@@ -0,0 +1,161 @@
+import java.lang.Math;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+
+public class PTUI {
+
+    private final int base;
+    private char[][] board;

```

Comment:
It makes more sense to have a Board object which contains the elements of the board and has methods like validMove, etc.